### PR TITLE
Created the Space Traveler's Hub Web Application. 

### PR DESCRIPTION
**Pull request by Asiimwe-Mark**
- We created the Space Traveler's Hub web application using React, react-redux.
- We implemented the Missions and Rockets pages by fetching the corresponding data from the provided APIs. 

We implemented all the requirements following the guidelines below.

# Config & basic setup
# Redux: Fetch data and update Redux store
# Render UI: lists
# Render UI: conditional components rendering
# Render UI: My Profile section 
# 
- Redux: Write actions and reducers for booking rockets/dragons and joining missions
- Redux: Write actions and reducers for canceling rockets/dragons and leaving missions

- We also added tests for all the relevant fields.

### CODE REVIEW

File: src/redux/missions/missionSlice.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    cancel: (state, action) => {
+      state.missions = state.missions.map((mission) => {
+        if (mission.mission_id === action.payload) {
+          return { ...mission, reserved: false };
+        }
+        return mission;
+      });
+    },
+  },
```
Comment (franclobo) [inline src/redux/missions/missionSlice.js:39]: - [ ] Please 🙏🏽 add tests for your functions reserve and cancel, it is a project requirement for this capstone 👇🏽 

> NOTE: Make sure to add tests for all your components and functions.

File: src/redux/rockets/rocketSlice.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    cancelRocket: (state, action) => {
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) {
+          return rocket;
+        }
+        return { ...rocket, reserved: false };
+      });
+    },
+  },
```
Comment (franclobo) [inline src/redux/rockets/rocketSlice.js:44]: - [ ] Please 🙏🏽 add tests for your functions reserve and cancel, it is a project requirement for this capstone 👇🏽 

> NOTE: Make sure to add tests for all your components and functions.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @As1imwe-Mark and @SnakyMz, 

# Suggested changes

istockphoto-1079018918-612x612

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Connect React and Redux ✔️
- Write unit tests with React Testing Library ✔️
- Use React hooks ✔️
- Use store, actions and reducers in React ✔️


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/redux/rockets/rocketSlice.js (hunk 2 above)
Comment (SnakyMz) [reply src/redux/rockets/rocketSlice.js:44]: Please any suggestions on how I can do the test.

File: src/components/MyMission.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            if (mission.reserved === true) {
+              return (
+                <tr key={mission.mission_id}>
+                  <td><h4>{mission.mission_name}</h4></td>
+                </tr>
+              );
+            }
+            return null;
+          })}
+        </tbody>
```
[the remark this answers is not in the document]
Comment (SnakyMz) [reply src/components/MyMission.js]: Screenshot 2023-08-31 005623
Changes as suggested

File: src/redux/missions/missionSlice.js (hunk 1 above)
Comment (SnakyMz) [reply src/redux/missions/missionSlice.js:39]: Screenshot 2023-08-31 005549
Changes as suggested

File: src/redux/rockets/rocketSlice.js (hunk 2 above)
Comment (SnakyMz) [reply src/redux/rockets/rocketSlice.js:44]: Screenshot 2023-08-31 005535
Changes as suggested

**Review by levy002 — APPROVED** [review]
## Hello @As1imwe-Mark & @SnakyMz 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter error :heavy_check_mark: 
- Implemented the suggested changes :heavy_check_mark: 

## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**